frontend/packages/enduser-frontend/src/views/applications/daycare-application/daycare-application.vue

```diff
@@ -1076,509 +1087,522 @@ SPDX-License-Identifier: LGPL-2.1-or-later
+import { DATE_FORMAT } from '@/constants'
+import FileUpload from '@/components/common/file-upload.vue'
+
+export default Vue.extend({
+  props: {
+    isPreschool: {
+      type: Boolean,
+      default: false
+    }
+  },
+  components: {
+    DaycareSelectList,
+    SortableUnitList,
+    // MapModal,
+    ChildInfo,
+    GuardianInfo,
+    Person,
+    Confirm,
+    ApplicationSummary,
+    CForm,
+    CTimeInput
+  },
+  data() {
+    return {
+      loading: false,
+      isDirty: false,
+      isChecked: false,
+      applicationLoaded: false,
+      activeSection: DAYCARE_SECTION.service,
+      model: this.isPreschool ? createPreschoolForm() : createDaycareForm(),
+      confirmRouteLeave: true,
+      showSummary: false,
+      validations: {},
+      sections: DAYCARE_SECTION,
+      errors: createDaycareSectionError(),
+      isValid: true,
+      summaryChecked: false,
+      guardianRequired: true,
+      language: [this.$t('constants.language.FI.value')],
+      debouncedValidate: _.debounce(() => false)
+    }
+  },
+  computed: {
+    ...mapGetters(['daycareForm', 'applicationUnits', 'urgentFiles', 'extendedCareFiles', 'originalPreferredStartDate']),
+    id(): string {
+      return this.$route.params.id
+    },
+    locale(): string {
+      return this.$i18n.locale
+    },
+    currentLocaleIsSwedish(): boolean {
+      return this.locale.toLowerCase() === LANGUAGES.SV
+    },
+    languages(): any {
+      return Object.values(
+        this.$t('constants.language', { returnObjects: true })
+      )
+    },
+    showDailyTime(): boolean {
+      return (
+        (this.isPreschool &&
+          (this.model as PreschoolFormModel).connectedDaycare) ||
+        !this.isPreschool
+      )
+    },
+    formattedStartDate(): string | null {
+      return this.model.preferredStartDate
+        ? formatDate(this.model.preferredStartDate, DATE_FORMAT)
+        : ''
+    },
+    minimumStartdate(): null | string {
+      // If application is already sent, the preferred start date can only be moved forward
+      return (this.model.status.value !== 'CREATED' && this.originalPreferredStartDate) ?
+        this.originalPreferredStartDate
+      : this.isPreschool
+        ? datepickerTodayFormat()
+        : minimumDaycareStartdate()
+    },
+    maximumStartDate(): null | string {
+      if(this.isPreschool && new Date() < new Date('2021-01-08')){
+        return '2021-06-04'
+      } else {
+        return null
       }
     },
-    components: {
-      DaycareSelectList,
-      SortableUnitList,
-      // MapModal,
-      ChildInfo,
-      GuardianInfo,
-      Person,
-      Confirm,
-      ApplicationSummary,
-      CForm,
-      CTimeInput
+    hasOutsidePreschoolTermError(): boolean {
+      const date = this.model.preferredStartDate ? new Date(this.model.preferredStartDate) : null
+      if(!this.isPreschool || !date) {
+        return false
+      }
+
+      return date >= new Date('2021-06-05') && date < new Date('2021-08-01')
```

Comment:
I assume this is just a temp hotfix? Maybe indicate that somehow here.

Reply:
Isn't the whole vue-based frontend a temporary hotfix? :D But yeah I'll add a comment.

Reply:
Yeah :D Just thinking about someone going over these validations during the rewrite